caiman/source_extraction/cnmf/utilities.py

```diff
@@ -45,7 +41,163 @@
+        return scipy.ndimage.uniform_filter(input, size, output, mode, cval)
+    size = np.atleast_1d(size).tolist()
+    if len(size)==1:
+        size = size*input.ndim
+    borderType = get_border_type(mode)
+
+    if input.ndim==2:
+        if mode=='constant' and cval!=0:
+            if output is None:
+                return cval+cv2.boxFilter(input-cval, -1, ksize=size[::-1], borderType=borderType)
+            else:
+                cv2.boxFilter(input-cval, -1, ksize=size[::-1], dst=output, borderType=borderType)
+                output += cval
+                return output
+        return cv2.boxFilter(input, -1, ksize=size[::-1], dst=output, borderType=borderType)
+
+    elif input.ndim==3:
+        if mode=='constant' and cval!=0:
+            input = input-cval
+        if output is None:
+            output = np.zeros_like(input)
+        for i in range(input.shape[1]):
+            output[:,i] = cv2.boxFilter(input[:,i], -1, ksize=size[2::-2], borderType=borderType)
+        if np.isfortran(input):
+            for i in range(input.shape[2]):
+                output[...,i] = cv2.boxFilter(output[...,i], -1, ksize=(size[1],1), borderType=borderType)
+        else:
+            for i in range(input.shape[0]):
+                output[i] = cv2.boxFilter(output[i], -1, ksize=(1,size[1]), borderType=borderType)
+        if mode=='constant' and cval!=0:
+            output+=cval
+        return output
+
+
+def maximum_filter(input, size=3, footprint=None, output=None, mode='reflect', cval=0.0):
+    if input.ndim>3 or (mode in ('constant', 'grid-constant') and cval!=0):
+        return ndi.maximum_filter(input, size, footprint, output, mode, cval)
+    size = np.atleast_1d(size).tolist()
+    if len(size)==1:
+        size = size*input.ndim if input.ndim>1 else [1]+size
+    borderType = get_border_type(mode)
+
+    if input.ndim==1:
+        return cv2.dilate(input[None,:], cv2.getStructuringElement(
+            cv2.MORPH_RECT, size[::-1]), borderType=borderType, 
+                             dst=None if output is None else output[None,:])[0]
+
+    elif input.ndim==2:
+        return cv2.dilate(input, cv2.getStructuringElement(
+            cv2.MORPH_RECT, size[::-1]), dst=output, borderType=borderType)
+
+    elif input.ndim==3:
+        if output is None:
+            output = np.zeros_like(input)
+        for i in range(input.shape[1]):
+            output[:,i] = cv2.dilate(input[:,i], cv2.getStructuringElement(
+            cv2.MORPH_RECT, size[2::-2]), borderType=borderType)
+        if np.isfortran(input):
+            for i in range(input.shape[2]):
+                output[...,i] = cv2.dilate(output[...,i], cv2.getStructuringElement(
+            cv2.MORPH_RECT, (size[1],1)), borderType=borderType)
+        else:
+            for i in range(input.shape[0]):
+                output[i] = cv2.dilate(output[i], cv2.getStructuringElement(
+            cv2.MORPH_RECT, (1,size[1])), borderType=borderType)
+        return output
```

Comment:
Same thing with the elif here

Reply:
added an else statement to the end falling back to the scipy.ndimage version